src/Cache/FileCache.php

```diff
@@ -47,7 +47,18 @@ public function put(ClassMetadata $metadata): void
         $path = $this->dir . '/' . strtr($metadata->name, '\\', '-') . '.cache.php';
 
         $tmpFile = tempnam($this->dir, 'metadata-cache');
-        file_put_contents($tmpFile, '<?php return unserialize(' . var_export(serialize($metadata), true) . ');');
+        if (false === $tmpFile) {
+            return;
+        }
+        $data = '<?php return unserialize(' . var_export(serialize($metadata), true) . ');';
+        $bytesWritten = file_put_contents($tmpFile, $data);
+        // use strlen and not mb_strlen. if there is utf8 in the code, it also writes more bytes.
+        if ($bytesWritten !== strlen($data)) {
+            // we did not write all characters, or completely failed to write.
+            @unlink($tmpFile);
+
+            return;
```

Comment:
on other failures, this class throws exceptions. being a cache, i would expect it to continue silently, as cache problems should not break the application. if you prefer, i can change this to an exception to be more consistent with the behaviour of the rest of the cache implementation.

Reply:
this should be a very bad situation but I guess it makes sense to cover this case...

When the disk is out of space... a lot of other bad things could happen...

Reply:
indeed so. we don't maintain the servers themselves and when we checked, there was ample disk space available. but with enough writes, one might fail for some reason somewhere, and in our case it left us with this half-written cache that broke everything (on one of several nodes only, so it looked like the error only happens randomly...)

Reply:
Alright. I guess this is pretty hard to test, right?

Reply:
the happy case is tested, but failure is difficult to test. maybe with stream wrappers and whatnot, but stream wrapper is so fragile that it randomly breaks other things too.